synapse/handlers/e2e_keys.py

```diff
@@ -977,12 +985,66 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
         Raises:
             NotFoundError: if the key is not found
         """
+        user = UserID.from_string(user_id)
         key = yield self.store.get_e2e_cross_signing_key(
             user_id, key_type, from_user_id
         )
+
+        if key is None and self.is_mine(user):
```

Comment:
Surely it should be:

```suggestion
        if key is None and not self.is_mine(user):
```

?